### ISSUE TITLE
Dependency on nq package

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/hap#18]
The following is displayed if HAP is loaded with OnlyNeeded option:
```
%%% Loading hap 1.22 , only needed
#I  HAP warning: Set POLYMAKE_PATH manually if needed. 
#I  HAP warning: Set ASY_PATH manually if needed.
Syntax warning: Unbound global variable in /circa/scratch/gap-jenkins/workspac\
e/GAP-pkg-update-stable-quicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packag\
esload/label/kovacs/GAP-pkg-update-stable-snapshot/pkg/hap-1.22/lib/Homology/i\
ntegralHomology.gi:150
FHhomH:=NqEpimorphismNilpotentQuotient(FH,1);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Syntax warning: Unbound global variable in /circa/scratch/gap-jenkins/workspac\
e/GAP-pkg-update-stable-quicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packag\
esload/label/kovacs/GAP-pkg-update-stable-snapshot/pkg/hap-1.22/lib/Homology/m\
odularHomology.gi:226
FHhomH:=NqEpimorphismNilpotentQuotient(FH,1);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Either NQ should be made a needed package, or some code should be loaded optionally.

**Comment by grahamknockillaree** [comment]
Fixed this on my local copy. Made nq a required package.

**Comment by alex-konovalov** [comment]
Thanks. Is this the sam problem or a different then: https://travis-ci.org/gap-infra/gap-docker-pkg-tests-stable-4.11/jobs/621170723 shows that the test fail when GAP is loaded with `-A` option:
```
########> Diff in /home/gap/inst/gap-stable-4.11/pkg/hap-1.22/tst/testall/hap.\
tst:21
# Input is:
R:=ResolutionAbelianGroup([0,0,0],5);;
# Expected output:
# But found:
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `*' on 2 arguments
########
```

**Comment by grahamknockillaree** [comment]
I've changed one line in Hap1.24/lib/Perturbations/resDirectProductLazy.gi to avoid  the product infinity*infinity . This product is undefined when using the gap -A  option.

But there is still an issue. Each test file works fine when called individually from GAP.  But when calling them all using Hap1.24/tst/testall.g  my laptop runs out of memory.

**Comment by alex-konovalov** [comment]
Looks to me like tests were running until 9 days ago: see https://travis-ci.org/gap-packages/hap/builds?utm_medium=notification&utm_source=github_status

**Comment by alex-konovalov** [comment]
One of possible ways to backtrack it would be to revert commit adding new tests to make them pass again, and then add tests files, possibly gradually, via pull requests. https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request explains how to create a pull request. A pull request will be tested on Travis automatically, then you can inspect whether tests pass or not, and only then decide whether to merge it or not. This will help you to keep the master branch in a clean state with passing tests.

**Comment by grahamknockillaree** [comment]
OK. I'll work on this.

Many of my tests require Polymake or Singular or other software. Probably this kind of test should be placed in a separate folder and only applied on my own laptop.

**Comment by alex-konovalov** [comment]
That will be a good way to ensure that we will never test them ;) - please don't rush doing that, I have another solution incoming.

**Comment by alex-konovalov** [comment]
@grahamknockillaree now let's see if #21 will fix things.

**Comment by alex-konovalov** [comment]
Resolved in 74cd7316baeaf42882e4da8b05cca6b026d9c154
